### PR TITLE
Add org.aegisub.Aegisub (aegisub-flatpak)

### PR DESCRIPTION
**Pull request by scx**
[Aegisub](http://www.aegisub.org) is a free, cross-platform open source tool for creating and modifying subtitles.

See also:
https://github.com/Aegisub/Aegisub
https://github.com/scx/aegisub-flatpak

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
I tried to create manifest for all supported arches. However, AArch64 is not officially supported and armhf may fail due to massive dependencies of the FFmpeg. To be honest, I tested only x86-64.

File: aegisub-flatpak-appdata.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+diff -Naur aegisub-flatpak.orig/org.aegisub.Aegisub.appdata.xml aegisub-flatpak/org.aegisub.Aegisub.appdata.xml
+--- aegisub-flatpak.orig/org.aegisub.Aegisub.appdata.xml	2018-12-06 03:18:01.115323000 +0100
++++ aegisub-flatpak/org.aegisub.Aegisub.appdata.xml	2018-12-06 19:30:41.135125735 +0100
+@@ -65,9 +65,8 @@
+   <url type="help">http://docs.aegisub.org</url>
+   <url type="homepage">http://www.aegisub.org</url>
+   <url type="translate">https://sites.google.com/site/rockytdrontransifex/aegisub</url>
+-  <content_rating type="oars-1.0">
+-    <content_attribute id="social-info">mild</content_attribute>
```
Comment (bilelmoussaoui) [inline aegisub-flatpak-appdata.patch]: Why this was removed?

File: boost.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+sources:
+  - type: archive
+    url: https://dl.bintray.com/boostorg/release/1.68.0/source/boost_1_68_0.tar.bz2
+    sha256: 7f6130bc3cf65f56a618888ce9d5ea704fa10b462be126ad053e80e553d6d8b7
+build-commands:
+  - ./bootstrap.sh --prefix="${FLATPAK_DEST}" --with-libraries=filesystem,locale,system,regex,thread,test;
+  - ./b2 headers;
+  - ./b2 install;
+post-install:
+  - install -p -D -m 0644 "LICENSE_1_0.txt" -t "${FLATPAK_DEST}/share/licenses/boost/";
```
Comment (bilelmoussaoui) [inline boost.yaml]: Not needed.

File: aom.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    arm:
+      config-opts:
+        - -DAOM_TARGET_CPU=generic
+sources:
+  - type: git
+    url: https://aomedia.googlesource.com/aom
+    branch: v1.0.0
+    commit: d14c5bb4f336ef1842046089849dee4a301fbbf0
+post-install:
+  - install -p -D -m 0644 "../LICENSE" "../PATENTS" -t "${FLATPAK_DEST}/share/licenses/aom/";
```
Comment (bilelmoussaoui) [inline aom.yaml]: License are not exported, shouldn't be installed.

File: ffmpeg.yaml [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+      config-opts:
+        - --enable-nvdec
+        - --enable-nvenc
+        - --enable-cuvid
+sources:
+  - type: archive
+    url: https://ffmpeg.org/releases/ffmpeg-4.1.tar.xz
+    sha256: a38ec4d026efb58506a99ad5cd23d5a9793b4bf415f2c4c2e9c1bb444acd1994
+post-install:
+  - install -p -D -m 0644 COPYING* "LICENSE.md" -t "${FLATPAK_DEST}/share/licenses/ffmpeg/";
```
Comment (bilelmoussaoui) [inline ffmpeg.yaml]: Same here for license.

File: aegisub-flatpak-appdata.patch (hunk 1 above)
Comment (scx) [reply aegisub-flatpak-appdata.patch]: > Why this was removed?

Not removed, but replaced. And I have already explained this in the comment.

https://github.com/scx/flathub/blob/d2d77a8d24dfc89ceea426b11ffa1d55baf5111b/Aegisub.yaml#L48-L51
```
  # Update AppData file
  # Change the OARS rating due to disabling the update checker
  - type: patch
    path: aegisub-flatpak-appdata.patch
```

See also:
https://github.com/scx/flathub/blob/d2d77a8d24dfc89ceea426b11ffa1d55baf5111b/Aegisub.yaml#L2-L3
```
config-opts:
  - --disable-update-checker
```

File: boost.yaml (hunk 2 above)
Comment (scx) [reply boost.yaml]: > Not needed.

You may not care about license compliance, but I and others do.
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L42
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L56
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L66
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L80
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L92
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L106
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L118
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L130
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L155
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L205
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L216
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L254
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L270
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L286
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L304
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L320
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L336
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L344
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L367
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L396
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L416
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L426
https://github.com/flathub/com.github.wwmm.pulseeffects/blob/3ca9162e35e63390ec3cbed2cb050e4e9e40dedc/com.github.wwmm.pulseeffects.yml#L439

See also:
https://github.com/flathub/flathub/pull/72#commits-pushed-f8778fe
```
While the legal questions of Flathub are being worked out this takes
some basic steps to ensure license compliance.

Most of the time, license compliance is as simple as including the
license and/or copyright file with distribution of the application.
In traditional linux packaging it is common to have a post install
step that does this. Fortunately flatpak-builder can do this too.
```

This is de facto required by law and I don't care whether these files are exported by flatpak or not - they must be included in the package.

File: ffmpeg.yaml (hunk 4 above)
Comment (scx) [reply ffmpeg.yaml]: > License are not exported, shouldn't be installed.

Encouraging breaking the license should not take place here.

File: aom.yaml (hunk 3 above)
Comment (scx) [reply aom.yaml]: > License are not exported, shouldn't be installed.

Encouraging breaking the license should not take place here.

File: ffmpeg.yaml (hunk 4 above)
Comment (scx) [reply ffmpeg.yaml]: > Same here for license.

As I said earlier, I'm not going to break the license.

**Comment by scx** [comment]
The package has been built successfully on arm, i386 and x86_64, but fails on aarch64, probably due to too old `config.{sub,guess}` for *twolame*.  I will try to fix it and then we'll see what's next.
https://flathub.org/builds/#/builders/2/builds/6571

**Comment by alexlarsson** [comment]
bot, build

**Comment by alexlarsson** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
Build fails on AArch64, this time probably due to built-in lua.
https://flathub.org/builds/#/builders/2/builds/6586
```
lj_arch.h:55:2: error: #error "No support for this architecture (yet)"
 #error "No support for this architecture (yet)"
```

I will try to add luajit module and patch Makefile.

**Comment by TingPing** [comment]
bot, build org.aegisub.Aegisub

**Comment by TingPing** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
It is successfully built on all architectures.
https://flathub.org/builds/#/builders/2/builds/6597

**Comment by nedrichards** [comment]
bot, build org.aegisub.Aegisub

**Comment by nedrichards** [comment]
testbot, build org.aegisub.Aegisub

**Comment by scx** [comment]
testbot, build org.aegisub.Aegisub

**Comment by scx** [comment]
testbot, build org.aegisub.Aegisub

**Comment by scx** [comment]
testbot, build org.aegisub.Aegisub

**Comment by scx** [comment]
testbot, build org.aegisub.Aegisub

**Comment by Cogitri** [comment]
Hey there, is there something left do be done here? Would be interested in helping here since Aegisub segfaults on musl distros :c

File: org.aegisub.Aegisub.yaml [hunk 5]
```diff
@@ -0,0 +1,81 @@
+app-id: org.aegisub.Aegisub
+runtime: org.gnome.Platform
+runtime-version: '3.30'
```
Comment (LeandroStanger) [inline org.aegisub.Aegisub.yaml]: Update runtime 3.32

**Comment by FakeShemp** [comment]
A lot of these modules seem unnecessary to me. How is OpenH264 and vid.stab used for example?

File: org.aegisub.Aegisub.yaml (hunk 5 above)
Comment (scx) [reply org.aegisub.Aegisub.yaml]: GNOME 3.30 was the latest release when I submitted this application (December 2018, over half a year ago!).
I don't see the point in updating it if it's not going to be merged anytime soon.
I can upgrade GNOME to 3.32, but that's it. Don't count on further updates if this PR will still be blocked.

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

File: org.aegisub.Aegisub.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  #- --socket=wayland
+  # Audio access
+  - --socket=pulseaudio
+  # Filesystem access
+  - --filesystem=host
+  # Dconf access
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --talk-name=ca.desrt.dconf
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
```
Comment (Cogitri) [inline org.aegisub.Aegisub.yaml]: I don't think this is needed anymore with https://blogs.gnome.org/mclasen/2019/07/12/settings-in-a-sandbox-world/?

**Comment by Cogitri** [comment]
Hm, FWIW Aegisub exits without any useful information for me, but so does the native version, so I guess there's just something Aegisub doesn't like about my system...

**Comment by scx** [comment]
@FakeShemp 

> A lot of these modules seem unnecessary to me.

These are FFmpeg dependencies.
You should know that I share the FFmpeg module between various applications. This is not an ideal situation, but maintaining many similar versions of FFmpeg seems to be completely unrealistic to me.
I wanted to create my own repository with shared modules, but this is forbidden here. It is also very unlikely that FFmpeg will be included in the Flathub [shared-modules](https://github.com/flathub/shared-modules) repo.

File: org.aegisub.Aegisub.yaml (hunk 6 above)
Comment (scx) [reply org.aegisub.Aegisub.yaml]: TingPing [said](https://blogs.gnome.org/mclasen/2019/07/12/settings-in-a-sandbox-world/#comment-2054):
> Only fdo 19.08 does it atm

However, GNOME 3.32 runtime is based on Freedesktop 18.08.

**Comment by scx** [comment]
@Cogitri 
As you can see, the Flathub maintainers are either too busy or just don't care. Unfortunately, there is nothing I can do about it. I've updated the GNOME runtime to the latest versions, but that's it.

I've provided a single-file bundle for x86_64.
https://github.com/scx/aegisub-flatpak/releases

It is slightly different from the version submitted here. Could you check if it works for you?

**Comment by FakeShemp** [comment]
@scx 
No I didn't know that :)

**Comment by Cogitri** [comment]
@scx Yay that works! Thank you very much!

Screenshot from 2019-08-10 21-54-24

**Comment by Cogitri** [comment]
Hm, would it maybe make sense to also force GDK_BACKEND to x11 due to Aegisub segfaulting otherwise[1][2]?

1: https://bugs.archlinux.org/task/55462
2: https://trac.wxwidgets.org/ticket/17702

**Comment by barthalion** [comment]
If reviewers' comments are addressed (including usual issues like unnecessary patches and excessive file splitting) and someone is interested in picking up maintenance, we'll merge it.

**Comment by scx** [comment]
@Cogitri 

> Hm, would it maybe make sense to also force GDK_BACKEND to x11 due to Aegisub segfaulting otherwise[1][2]?
> 
> 1: https://bugs.archlinux.org/task/55462
> 2: https://trac.wxwidgets.org/ticket/17702

I perfectly know about the Aegisub issue with Wayland. That's why I've disabled Wayland access last year.
https://github.com/scx/flathub/commit/160ce7d6218601c217e64144b93109508b799d10
Moreover, I tested it on Fedora (~27-29) and it worked well.

I don't know why flatpak wants to use Wayland when I explicitly forced X11.

Today I tried to force `GDK_BACKEND` to `X11`, but it doesn't work either.
```
finish-args:
  - --env=GDK_BACKEND=X11
```
```
Error: Unable to initialize gtk, is DISPLAY set properly?
```
At the moment I have no more ideas, how I can solve this problem.

Maybe we should switch to wxGTK2. This should be good because it also solves minority issues with UI that occur with wxGTK3. However, Flathub maintainers insist on using Gtk+3 whenever possible.

File: org.aegisub.Aegisub.yaml (hunk 6 above)
Comment (Cogitri) [reply org.aegisub.Aegisub.yaml]: Oh alright, thanks for the info.

**Comment by Cogitri** [comment]
> I don't know why flatpak wants to use Wayland when I explicitly forced X11.

Oh, I'm very sorry, I don't use Wayland currently and just assumed that it's broken on Wayland (since I've experienced that previously when using Aegisub's native package a few years ago when I used Arch w/ Wayland) and just wanted to make sure you knew about that since I couldn't find references to GDK_BACKEND in the flatpak. I didn't think about the possibility of just denying access to Wayland, that may very well work! As such I think it's fine to use GTK+3 :)

Thanks for your work! :)

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

**Comment by LeandroStanger** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
The last commits should improve stability and fix many itches, including these ones:
- https://github.com/Aegisub/Aegisub/issues/110
- https://github.com/Aegisub/Aegisub/issues/121#issuecomment-495214637
- https://github.com/Aegisub/Aegisub/issues/131
- https://github.com/Aegisub/Aegisub/issues/133
- https://github.com/flatpak/flatpak/issues/1097#issuecomment-527930192
- https://github.com/flatpak/flatpak/issues/2134
- Broken spell checking
- Various issues with Unicode and CJK/emoji support
- Various issues with **Automation** scripts/`LUA`

**Comment by Cogitri** [comment]
Gonna test in a bit, thanks for working on this

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

**Comment by scx** [comment]
bot, build org.aegisub.Aegisub

**Comment by barthalion** [comment]
Given Bilel's review wasn't addressed for more than a year, it's safe to assume nothing is going to change about it.

**Pull request closed without merging**